utilities/src/main/java/io/syndesis/qe/utils/TestUtils.java

```diff
@@ -158,4 +161,47 @@ public static void sleepIgnoreInterrupt(long milis) {
             e.printStackTrace();
         }
     }
+
+    public static void setDatabaseCredentials(String connectionName, DBAllocation dbAllocation) {
+
+        Map<String, String> allocPropertiesMap = dbAllocation.getAllocationMap();
+
+        Optional<Account> optional = AccountsDirectory.getInstance().getAccount(connectionName);
+
+        Account dbAccount;
+
+//        if (optional.isPresent()) {
+//            TestUtils.setAccountProperties(dbAccount, allocProperties);
+//        } else {
+//            Assertions.fail("There is no " + connectionName + " input in your credential json!");
+//        }
+
+        if (!optional.isPresent()) {
+            dbAccount = new Account();
+            dbAccount.setService(connectionName);
+//            dbAccount.setProperties(new HashMap<String, String>());
+        } else {
+            dbAccount = optional.get();
+        }
+        TestUtils.setAccountProperties(dbAccount, allocPropertiesMap);
+    }
+
+    private static void setAccountProperties(Account account, Map<String, String> sourceMap) {
```

Comment:
Yes you made this method more universal, but unless service name of the account is "oracle12" or "mysql", method does nothing. So in fact it is not universal at all. To make it universal you have to add default case to your switch and put whole source map into account properties.

Reply:
The idea was to handle dynamically generated accounts. Yes, for the time being there are only SQL Databases allocated dynamically, which can change in the future, so other cases will be added on need. 
Account properties key values related to fill-form, since this method serves rather as transshipment place,  I prefer to keep it as it is.

Reply:
comments incorporated & pushed